### PR TITLE
Xcode 8.3 support

### PR DESCRIPTION
**Pull request by mbrandonw**
Adds support for Xcode 8.3 and Swift 3.1. Same sitch as https://github.com/kickstarter/ios-ksapi/pull/21, https://github.com/kickstarter/Kickstarter-Prelude/pull/70 and https://github.com/kickstarter/Kickstarter-ReactiveExtensions/pull/61.

One cool thing about Xcode 8.3... playgrounds work again! We can get back to our old Xcode 7.x flow of using playgrounds for UI dev.

### Todo

- [x] circleci xcode 8.3 support
- [x] figure out `static func initialize` stuff

### CODE REVIEW

File: Kickstarter-iOS/Views/ReferralChartView.swift [hunk 1]
```diff
@@ -30,17 +30,17 @@ public final class ReferralChartView: UIView {
 
     guard let context = UIGraphicsGetCurrentContext() else { return }
 
-    let internalPercentageAngle = CGFloat(-M_PI_2) + self.internalPercentage * CGFloat(2.0 * M_PI)
+    let internalPercentageAngle = CGFloat(-.pi / 2.0) + self.internalPercentage * CGFloat(2.0 * .pi)
```
Comment (mbrandonw) [inline Kickstarter-iOS/Views/ReferralChartView.swift]: this is a fun one! `pi` has been moved to a static var on `Double`, and the type-inference makes it really niiiiice

File: Library/AppEnvironment.swift [hunk 2]
```diff
@@ -118,7 +118,7 @@ public struct AppEnvironment {
     locale: Locale = AppEnvironment.current.locale,
     mainBundle: NSBundleType = AppEnvironment.current.mainBundle,
     reachability: SignalProducer<Reachability, NoError> = AppEnvironment.current.reachability,
-    scheduler: DateSchedulerProtocol = AppEnvironment.current.scheduler,
+    scheduler: DateScheduler = AppEnvironment.current.scheduler,
```
Comment (mbrandonw) [inline Library/AppEnvironment.swift]: just a rename from ReactiveSwift 1.1.0

File: Library/DataSource/UIView-Extensions.swift [hunk 3]
```diff
@@ -25,6 +25,8 @@ private func swizzle(_ v: UIView.Type) {
 }
 
 extension UIView {
+  // FIXME: "Method 'initialize' defines Objective-C class method, which is not guaranteedto be invoked by 
+  // by Swift and will be disallowed in future versions.
   open override class func initialize() {
```
Comment (mbrandonw) [inline Library/DataSource/UIView-Extensions.swift]: gotta figure this one out before merging :/ still thinking about possible solutions

File: Library/ViewModels/DeprecatedWebViewModel.swift [hunk 4]
```diff
@@ -41,7 +41,7 @@ DeprecatedWebViewModelOutputs {
       // Hide loading if the web view fails with a non-102 error code (102 is the interrupted error that
       // occurs anytime we cancel a request).
       self.webViewDidFailErrorProperty.signal
-        .filter { ($0 as? NSError)?.code != .some(102) }
+        .filter { ($0 as NSError?)?.code != .some(102) }
```
Comment (mbrandonw) [inline Library/ViewModels/DeprecatedWebViewModel.swift]: don't know why this works, but whatever it does. feel there is some error error/nserror bridging going on, and not sure i care enough to know why...

File: Kickstarter-iOS/Views/ReferralChartView.swift (hunk 1 above)
Comment (justinswart) [inline Kickstarter-iOS/Views/ReferralChartView.swift]: ah nice!

File: Library/DataSource/UIView-Extensions.swift (hunk 3 above)
Comment (justinswart) [inline Library/DataSource/UIView-Extensions.swift]: Perhaps just function calls from `AppDelegate` `didFinishLaunchingWithOptions`?

Comment (mbrandonw) [inline Library/DataSource/UIView-Extensions.swift]: 👍 going that route

File: Library/ViewModels/DeprecatedWebViewModel.swift (hunk 4 above)
Comment (mbrandonw) [inline Library/ViewModels/DeprecatedWebViewModel.swift]: ah yes that makes sense, forgot that `$0` was optional at that moment

File: Kickstarter-iOS/AppDelegate.swift [hunk 5]
```diff
@@ -30,6 +30,9 @@ internal final class AppDelegate: UIResponder, UIApplicationDelegate {
     _ application: UIApplication,
     didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
 
+    UIView.doBadSwizzleStuff()
+    UIViewController.doBadSwizzleStuff()
```
Comment (mbrandonw) [inline Kickstarter-iOS/AppDelegate.swift]: welp here's where we're at. doing our bad swizzle stuff first thing when the app launches.

at least we are being honest now and not hiding deep somewhere else!

File: Library/TestHelpers/TestCase.swift [hunk 6]
```diff
@@ -28,6 +28,9 @@ internal class TestCase: FBSnapshotTestCase {
   override func setUp() {
     super.setUp()
 
+    UIView.doBadSwizzleStuff()
+    UIViewController.doBadSwizzleStuff()
```
Comment (mbrandonw) [inline Library/TestHelpers/TestCase.swift]: oh also we have to do it tests for screenshots :P

**Pull request merged**